packages/core/src/declareAtom.ts

```diff
@@ -160,66 +140,64 @@ export function declareAtom<State>(
   return atom
 }
 
-export function getState<T>(state: State, atom: Atom<T>): T | undefined {
+export function getState<T>(state: State, atom: Atom<T>): T {
```

Comment:
getState and store.getState is different functions. The getState must return optional type